snmp/datadog_checks/snmp/config.py

```diff
@@ -168,28 +234,59 @@ def get_context_data(instance):
 
         return context_engine_id, context_name
 
+    def resolve_oid(self, oid):
```

Comment:
If that's possible, I think it would be better to put the resolver (resolver/OIDTrie/OIDTreeNode) in a separate file/class e.g. `resolver.py`. That would make it easier to read/maintain.

The `config.py/ParsedTableMetric` seems quite complex already.

Reply:
Good idea, trying.